core/geometry/src/curve/internalContexts/EllipticalArcApproximationContext.ts

```diff
@@ -0,0 +1,1106 @@
+   * @return quadrant number and start/end radian angles for the quadrant that contains both input angles, or
+   * undefined if no such quadrant.
+   * * The returned sweep is always counterclockwise: angle0 < angle1.
+   */
+  public static getQuadrantRadians(
+    radians0: number, radians1: number,
+  ): { quadrant: 1 | 2 | 3 | 4, angle0: number, angle1: number } | undefined {
+    if (AngleSweep.isRadiansInStartEnd(radians0, 0, Angle.piOver2Radians)
+      && AngleSweep.isRadiansInStartEnd(radians1, 0, Angle.piOver2Radians))
+      return { quadrant: 1, angle0: 0, angle1: Angle.piOver2Radians };
+    if (AngleSweep.isRadiansInStartEnd(radians0, Angle.piOver2Radians, Angle.piRadians)
+      && AngleSweep.isRadiansInStartEnd(radians1, Angle.piOver2Radians, Angle.piRadians))
+      return { quadrant: 2, angle0: Angle.piOver2Radians, angle1: Angle.piRadians };
+    if (AngleSweep.isRadiansInStartEnd(radians0, Angle.piRadians, Angle.pi3Over2Radians)
+      && AngleSweep.isRadiansInStartEnd(radians1, Angle.piRadians, Angle.pi3Over2Radians))
+      return { quadrant: 3, angle0: Angle.piRadians, angle1: Angle.pi3Over2Radians };
+    if (AngleSweep.isRadiansInStartEnd(radians0, Angle.pi3Over2Radians, Angle.pi2Radians)
+      && AngleSweep.isRadiansInStartEnd(radians1, Angle.pi3Over2Radians, Angle.pi2Radians))
+      return { quadrant: 4, angle0: Angle.pi3Over2Radians, angle1: Angle.pi2Radians };
+    return undefined;
+  }
+};
+
+/**
+ * Base class for processing samples of an elliptical arc.
+ * @internal
+ */
+abstract class QuadrantFractionsProcessor {
+  /**
+   * Announce the beginning of processing of the samples in a given quadrant.
+   * @return whether to process this quadrant
+  */
+  public announceQuadrantBegin(_q: QuadrantFractions): boolean { return true; }
+  /**
+   * Optionally announce a chord of the elliptical arc between the given fractions.
+   * @param pt0 start point of the chord segment
+   * @param pt1 end point of the chord segment
+   * @param f0 elliptical arc parameter in [0,1] of pt0
+   * @param f1 elliptical arc parameter in [0,1] of pt1
+   */
+  public announceChord?(pt0: Point3d, pt1: Point3d, f0: number, f1: number): void;
+  /**
+   * Optionally announce a circular arc approximating the elliptical arc between the given fractions.
+   * @param arc circular arc interpolating the ellipse at the given fractions.
+   * * The processor is allowed to capture `arc`, e.g., to add it to a growing chain approximation.
+   * @param f0 elliptical arc parameter in [0,1], unordered with respect to f1
+   * @param f1 elliptical arc parameter in [0,1], unordered with respect to f0
+   */
+  public announceArc?(arc: Arc3d, f0: number, f1: number): void;
+  /**
+   * Announce the end of processing of the samples in the given quadrant.
+   * @param _wasReversed whether the given quadrant's samples were processed in reverse
+  */
+  public announceQuadrantEnd(_q: QuadrantFractions, _wasReversed: boolean): void { }
```

Comment:
I move this a few lines above right after `announceQuadrantBegin`

Reply:
It's there because the intervening methods occur between announceBegin/End.